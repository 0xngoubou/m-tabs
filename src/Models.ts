export namespace Models {

    export interface TabData {
        key?: string;
        title: string;
        /** for user's custom extends */
        [key: string]: any;
    }

}