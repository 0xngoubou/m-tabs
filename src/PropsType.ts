import { Models } from './Models';

export interface TabBarPropsType {
    /** 跳转Tab */
    goToTab: (index: number) => void;
    /** tab数据 */
    tabs: Models.TabData[];
    /** 当前激活Tab索引 */
    activeTab: number;
    /** 是否使用动画 */
    animated: boolean;
    /** TabBar位置 | top: 上, bottom: 下 */
    tabBarPosition?: 'top' | 'bottom'; // TODO left, right

    // TabBar shortcut settings.
    tabBarUnderlineStyle?: React.CSSProperties;
    tabBarBackgroundColor?: string;
    tabBarActiveTextColor?: string;
    tabBarInactiveTextColor?: string;
    tabBarTextStyle?: React.CSSProperties;
}

export interface PropsType {
    /** 样式前缀(web only) | default: rmc-tabs */
    prefixCls?: string;
    /** tab数据 */
    tabs: Models.TabData[];
    /** TabBar位置 | default: top | top: 上, bottom: 下 */
    tabBarPosition?: 'top' | 'bottom'; // TODO left, right
    /** 替换TabBar */
    renderTabBar?: (props: TabBarPropsType) => React.ReactNode;
    /** 初始化Tab, index or key */
    initalPage?: number | string;
    /** 当前Tab, index or key */
    page?: number | string;
    /** 是否可以滑动内容切换 | default: true */
    swipeable?: boolean;
    /** 预加载两侧Tab数量 | default: 1 */
    prerenderingSiblingsNumber?: number;
    /** 是否开启切换动画 | default: true */
    animated?: boolean;
    /** tab变化时触发 */
    onChangeTab?: (index: number, tabData: Models.TabData) => void;
    /** 使用跟手滚动 | default: true */
    useOnPan?: boolean;

    // TabBar shortcut settings.
    tabBarUnderlineStyle?: React.CSSProperties;
    tabBarBackgroundColor?: string;
    tabBarActiveTextColor?: string;
    tabBarInactiveTextColor?: string;
    tabBarTextStyle?: React.CSSProperties;
}