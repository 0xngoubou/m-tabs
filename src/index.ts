export { Models } from './Models';
export { Tabs } from './Tabs';
export { DefaultTabBar } from './DefaultTabBar';