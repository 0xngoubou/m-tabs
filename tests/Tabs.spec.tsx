import React from 'react';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import { Tabs } from '../src';

const tabData = [
    { key: 't1', title: 't1' },
    { key: 't2', title: 't2' },
    { key: 't3', title: 't3' },
    { key: 't4', title: 't4' },
    { key: 't5', title: 't5' },
];
const tabDataWithoutKey = [
    { title: 't1' },
    { title: 't2' },
    { title: 't3' },
    { title: 't4' },
    { title: 't5' },
];

const renderContent = () => {
    const pStyle = { margin: 0, padding: 10 } as React.CSSProperties;

    return [
        <div key="t1" style={{ background: '#ADFFD7' }}>
            <p style={pStyle}>tab 1 1</p>
            <p style={pStyle}>tab 1 2</p>
            <p style={pStyle}>tab 1 3</p>
            <p style={pStyle}>tab 1 4</p>
        </div>,
        <div key="t2" style={{ background: '#ADFFD7' }}>
            <p style={pStyle}>tab 2 1</p>
            <p style={pStyle}>tab 2 2</p>
            <p style={pStyle}>tab 2 3</p>
            <p style={pStyle}>tab 2 4</p>
            <p style={pStyle}>tab 2 5</p>
        </div>,
        <div key="t3" style={{ background: '#ADFFD7' }}>tab 3</div>,
        <div key="t4" style={{ background: '#ADFFD7' }}>tab 4</div>,
        <div key="t5" style={{ background: '#ADFFD7' }}>tab 5</div>,
    ];
}

describe('basic', () => {
    it('base.', () => {
        const wrapper = render(
            <Tabs tabData={tabData}
            >
                {renderContent()}
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('no animation.', () => {
        const wrapper = render(
            <Tabs tabData={tabData}
                animated={false}
            >
                {renderContent()}
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('inital tab.', () => {
        const wrapper = render(
            <Tabs tabData={tabData}
                initalTab={'t2'}
            >
                {renderContent()}
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('tabBarPosition.', () => {
        const wrapper = render(
            <Tabs tabData={tabData}
                tabBarPosition="bottom"
            >
                {renderContent()}
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('prerenderingSiblingsNumber.', () => {
        const wrapper = render(
            <Tabs tabData={tabData}
                prerenderingSiblingsNumber={1}
            >
                {renderContent()}
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });
});

describe('single content.', () => {
    it('function', () => {
        const wrapper = render(
            <Tabs tabData={tabDataWithoutKey}
            >
                {
                    (index, tab) =>
                        <div>
                            <p>single content</p>
                            <p>{JSON.stringify({ index, tab })}</p>
                        </div>
                }
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });

    it('base', () => {
        const wrapper = render(
            <Tabs tabData={tabDataWithoutKey}
            >
                <div>
                    <p>single content</p>
                </div>
            </Tabs>
        );
        expect(renderToJson(wrapper)).toMatchSnapshot();
    });
});