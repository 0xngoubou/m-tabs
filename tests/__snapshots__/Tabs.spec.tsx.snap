// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic base. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic destroyInactiveTab. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:40%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(-200%, 0px, 0px);-webkit-transform:translate3d(-200%, 0px, 0px);-moz-transform:translate3d(-200%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        tab 3
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic inital tab. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:20%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(-100%, 0px, 0px);-webkit-transform:translate3d(-100%, 0px, 0px);-moz-transform:translate3d(-100%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        tab 3
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic no animation. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic prerenderingSiblingsNumber. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic renderTabBar renderTab. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          <div>
            t1
          </div>
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          <div
            style="position:relative;"
          >
            t2
            <div
              style="position:absolute;right:-3px;top:-1px;width:6px;height:6px;background:red;border-top-left-radius:6px;border-bottom-left-radius:6px;border-bottom-right-radius:6px;border-top-right-radius:6px;"
            />
          </div>
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          <div>
            t3
          </div>
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          <div>
            t4
          </div>
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          <div>
            t5
          </div>
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`basic tabBarPosition. 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-bottom"
>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-bottom"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`basic vertical. 1`] = `
<div
  class="rmc-tabs rmc-tabs-vertical rmc-tabs-tabbar-left"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-left"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="height:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="height:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="height:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="height:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="height:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="height:20%;top:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-x;transform:translate3d(0px, 0%, 0px);-webkit-transform:translate3d(0px, 0%, 0px);-moz-transform:translate3d(0px, 0%, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 1 4
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div
        style="background:#ADFFD7;"
      >
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 1
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 2
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 3
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 4
        </p>
        <p
          style="margin:0;padding:10px;"
        >
          tab 2 5
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`single content. base 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div>
        <p>
          single content
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div>
        <p>
          single content
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;

exports[`single content. function 1`] = `
<div
  class="rmc-tabs rmc-tabs-horizontal rmc-tabs-tabbar-top"
>
  <div
    class="rmc-tabs-tab-bar-wrap"
  >
    <div
      class="rmc-tabs-tab-bar rmc-tabs-tab-bar-animated rmc-tabs-tab-bar-top"
      style="background-color:#fff;"
    >
      <div
        class="rmc-tabs-tab-bar-content"
        style="touch-action:pan-x pan-y;"
      >
        <div
          class="rmc-tabs-tab-bar-tab rmc-tabs-tab-bar-tab-active"
          style="width:20%;"
        >
          t1
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t2
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t3
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t4
        </div>
        <div
          class="rmc-tabs-tab-bar-tab"
          style="width:20%;"
        >
          t5
        </div>
        <div
          class="rmc-tabs-tab-bar-underline"
          style="width:20%;left:0%;"
        />
      </div>
    </div>
  </div>
  <div
    class="rmc-tabs-content-wrap rmc-tabs-content-wrap-animated"
    style="touch-action:pan-y;transform:translate3d(0%, 0px, 0px);-webkit-transform:translate3d(0%, 0px, 0px);-moz-transform:translate3d(0%, 0px, 0px);"
  >
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-active"
    >
      <div>
        <p>
          single content
        </p>
        <p>
          {"index":0,"tab":{"title":"t1"}}
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    >
      <div>
        <p>
          single content
        </p>
        <p>
          {"index":1,"tab":{"title":"t2"}}
        </p>
      </div>
    </div>
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
    <div
      class="rmc-tabs-pane-wrap rmc-tabs-pane-wrap-inactive"
    />
  </div>
</div>
`;
